protostar-rust/src/main.rs

```diff
@@ -66,16 +77,38 @@ fn dependencies_for_package(
 fn main_execution() -> Result<()> {
     let _args = Args::parse();
 
+    let project_root_path = project_root::get_project_root()?
+        .to_str()
+        .expect("error reading project root path")
+        .to_owned();
     // TODO #1997
-    set_var("CARGO_MANIFEST_DIR", "../../cairo/Cargo.toml");
+    set_var(
+        "CARGO_MANIFEST_DIR",
+        format!("{}/../cairo/Cargo.toml", project_root_path),
+    );
 
     let scarb_metadata = MetadataCommand::new().inherit_stderr().exec()?;
 
+    let mut protostar_test_config = ProtostarTestConfig {
+        contracts_paths: HashMap::new(),
+    };
+    for package in &scarb_metadata.packages {
+        for target in &package.targets {
+            if target.kind == "starknet-contract" {
+                // TODO consider multiple targets of this kind
+                protostar_test_config
+                    .contracts_paths
+                    .insert(target.name.clone(), target.source_path.to_string());
+            }
+        }
```

Comment:
This info shouldn't be loaded from metadata, we should run scarb build to build contracts and use produced json artifact instead. It will be simpler to reason about and we will have less code on our side.